x/sp/keeper/msg_server.go

```diff
@@ -82,6 +83,11 @@ func (k msgServer) CreateStorageProvider(goCtx context.Context, msg *types.MsgCr
 		return nil, types.ErrStorageProviderGcAddrExists
 	}
 
+	// check to see if the test address has been registered before
+	if _, found := k.GetStorageProviderByTestAddr(ctx, testAcc); found {
```

Comment:
I think we do not have to check this and  support query SP by testaddr,  it is ok to have two SP share a same testaddr, it does not matter.